core/model/src/main/java/org/eclipse/rdf4j/model/impl/SimpleLiteral.java

```diff
@@ -54,6 +54,12 @@
 	 */
 	private IRI datatype;
 
+	// The XSD.Datatype enum that matches the datatype IRI for this literal. This value is calculated on the fly and
+	// cached in this variable. `null` means we have not calculated and cached this value yet. We are not worried about
+	// race conditions, since calculating this value multiple times must lead to the same effective result. Transient is
+	// only used to stop this field from be serialised.
+	transient private Optional<XSD.Datatype> xsdDatatype;
```

Comment:
It's a little odd to be using an Optional value that can also be `null`. I understand why it's done (and the comment also explains it well), but to make it glaringly obvbious perhaps initialize the value as `null` here as well, explicitly.